yt_dlp/extractor/twitcasting.py

```diff
@@ -79,74 +93,122 @@ def _real_extract(self, url):
         title = (clean_html(get_element_by_id('movietitle', webpage))
                  or self._html_search_meta(['og:title', 'twitter:title'], webpage, fatal=True))
 
-        video_js_data = {}
-        m3u8_url = self._search_regex(
-            r'data-movie-url=(["\'])(?P<url>(?:(?!\1).)+)\1',
-            webpage, 'm3u8 url', group='url', default=None)
-        if not m3u8_url:
-            video_js_data = self._parse_json(self._search_regex(
-                r'data-movie-playlist=(["\'])(?P<url>(?:(?!\1).)+)',
-                webpage, 'movie playlist', group='url', default='[{}]'), video_id)
-            if isinstance(video_js_data, dict):
-                video_js_data = list(video_js_data.values())[0]
-            video_js_data = video_js_data[0]
-            m3u8_url = try_get(video_js_data, lambda x: x['source']['url'])
-
-        stream_server_data = self._download_json(
-            'https://twitcasting.tv/streamserver.php?target=%s&mode=client' % uploader_id, video_id,
-            'Downloading live info', fatal=False)
-
-        is_live = 'data-status="online"' in webpage
-
-        if not traverse_obj(stream_server_data, 'llfmp4') and is_live:
-            self.raise_login_required(method='cookies')
+        video_js_data = try_get(
+            webpage,
+            lambda x: self._parse_json(self._search_regex(
+                r'data-movie-playlist=\'([^\']+?)\'',
+                x, 'movie playlist', default=None), video_id)['2'], list)
 
-        formats = []
-        if is_live and not m3u8_url:
-            m3u8_url = 'https://twitcasting.tv/%s/metastream.m3u8' % uploader_id
-        if is_live and has_websockets and stream_server_data:
-            qq = qualities(['base', 'mobilesource', 'main'])
-            streams = traverse_obj(stream_server_data, ('llfmp4', 'streams')) or {}
-            for mode, ws_url in streams.items():
-                formats.append({
-                    'url': ws_url,
-                    'format_id': 'ws-%s' % mode,
-                    'ext': 'mp4',
-                    'quality': qq(mode),
-                    'protocol': 'websocket_frag',  # TwitCasting simply sends moof atom directly over WS
-                })
-
-        thumbnail = video_js_data.get('thumbnailUrl') or self._og_search_thumbnail(webpage)
+        thumbnail = try_get(
+            video_js_data,
+            (lambda x: traverse_obj(x, (0, 'thumbnailUrl')),
+             lambda x: self._og_search_thumbnail(webpage)),
+            str)
```

Comment:
```suggestion
        thumbnail = traverse_obj(video_js_data, (0, 'thumbnailUrl')) or self._og_search_thumbnail(webpage)
```
why try_get?

Reply:
it used to be just indexing where `try_get` is applicable